### PR TITLE
Fix all XML files detected as SVG

### PR DESCRIPTION
**Pull request by jcjgraf**
All XML files were detected as SVGs. This bug has been in vimiv ever since SVGs are supported (441b2d). Stupidly, I just copied to old SVG check function in #650 without thinking 😬

A xml file (i.e. a file starting with `<?xml`) is a svg, if and only if it is followed by a `<svg` tag[1].

[1] Do not quote me on this, standards are weird and the signature differentiates between different SVG standard versions. It is likely that we still have false negative rate, but we better only extend the detected signatures once we encounter such samples out in the wild...

### CODE REVIEW

File: vimiv/utils/imageheader.py [hunk 1]
```diff
@@ -171,26 +184,44 @@ def _test_gif(h: bytes, _f: BinaryIO) -> bool:
...
+    if h[2:5] == b"\x73\x76\x67":
+        return True
+
+    # Check for svg start tag
+    import re
+
+    f.seek(0, 0)
+    h = f.read()
+    pattern = b"<svg"
+    return re.search(pattern, h) is not None
```
Comment (karlch) [inline vimiv/utils/imageheader.py:224]: I suppose we need the full-blown re search, as there can be all sorts of stuff in front (comments, newlines, spacing, <?xml, ...)?

File: docs/changelog.rst [hunk 2]
```diff
@@ -6,6 +6,11 @@ All notable changes to vimiv are documented in this file.
 v0.10.0 (unreleased)
 --------------------
 
+Fixed:
+^^^^^^
+
+* The image header check for SVG was faulty and returned true for all XML files.
```
Comment (karlch) [inline docs/changelog.rst:12]: This sounds a bit harsh, but feel free to keep :blush: Would prefer something along:
Improved the SVG image header check to no longer return true for all XML files.

**Review by karlch — COMMENTED** [review]
Thanks for finding this funky behaviour! Left a small question, but am basically happy to merge as-is.

File: vimiv/utils/imageheader.py (hunk 1 above)
Comment (jcjgraf) [reply vimiv/utils/imageheader.py:224]: A full search opposed to a strict matching (i.e. something like `<?xml>\n<svg>`) you mean? I presume that this does not not work in all cases. I have also seen SVGs that have multiple `<svg` tag. Therefor, I thought that there may be also some other xml tags in front of it, or that the `<svg` tag is at a higher depth. But I am by no means a specialist...

File: docs/changelog.rst (hunk 2 above)
Comment (jcjgraf) [reply docs/changelog.rst]: I have improved the wording

File: vimiv/utils/imageheader.py (hunk 1 above)
Comment (jcjgraf) [reply vimiv/utils/imageheader.py:224]: Once this is resolved, the PR is mergeable.

Comment (karlch) [reply vimiv/utils/imageheader.py:224]: Finding them is definitely a mess, I guess using `re` is the way to go, a random file I had around e.g. started with:
```
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
  "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Created with matplotlib (https://matplotlib.org/) -->
<svg height="255pt" version="1.1" viewBox="0 0 660 255" width="660pt" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```

**Pull request merged**